sys/include/bitfield.h

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (C) 2015 INRIA
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    sys_bitfield bitfields of arbitrary length
+ * @ingroup     sys
+ * @brief       bitfields of arbitrary length
+ * @{
+ *
+ * @brief       bitfields operations on bitfields of arbitrary length
+ *
+ * @note        Code taken mostly from
+ *              <a href="http://stackoverflow.com/questions/1590893/error-trying-to-define-a-1-024-bit-128-byte-bit-field">
+ *              Stackoverflow, User Christoph</a>
+ *
+ * @author      Oliver Hahm <oliver.hahm@inria.fr>
+ */
+
+#ifndef BITFIELD_H
+#define BITFIELD_H
+
+#include <limits.h>
+#include <stdbool.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Declare a bitfield of a given size
+ *
+ * @note    SIZE should be a constant expression this avoids variable length
+ *          arrays and problems resulting from being evaluated twice
+ */
+#define BITFIELD(SIZE, NAME) \
```

Comment:
I think NAME, SIZE is a more intuitive order of arguments, like in `uint8_t NAME[SIZE]`